cpp/src/robotick/framework/data/DataConnection.cpp

```
@@ -190,6 +190,10 @@ namespace robotick
 			if (!field)
 				ROBOTICK_FATAL_EXIT("Field '%s' not found in path: %s", field_token.c_str(), path);
 
+			// get_data_ptr sums the instance base address and the section offset computed during Engine::load(), so every
+			// field pointer stays stable even as other workloads expand.  That offset math mirrors the contiguous layout.
+			// get_data_ptr sums the engine's contiguous base pointer with the struct offset computed during Engine::load(),
+			// so every resolved pointer matches the deterministic layout even though the memory lives in a single buffer.
 			const uint8_t* ptr = (uint8_t*)field->get_data_ptr(workloads_buffer, *workload, *struct_type, struct_offset);
```

Clang (14.0.6)

[warning] variable 'ptr' is not initialized

(cppcoreguidelines-init-variables)